src/nodedef.cpp

```diff
@@ -561,11 +569,24 @@ void ContentFeatures::deSerialize(std::istream &is, u16 protocol_version)
 		int value = readS16(is);
 		groups[name] = value;
 	}
-	param_type = (enum ContentParamType) readU8(is);
-	param_type_2 = (enum ContentParamType2) readU8(is);
+	
+	u8 tmp = readU8(is);
+	if (tmp > 1)
```

Comment:
this isn't very future-proof: If the enum is extended in the future this has to be changed too and you have to find out all places where this is the case.

Reply:
What would you prefer?
- an additional dummy value in the enum (at the end) to check for the maximum element
- a macro for the enum length
- a switch case over all enum values (this gets very large (and may be slower) for ContentParamType2 and NodeDrawType)

Reply:
I suppose the first option is easiest/best

Reply:
:+1: for the maximum value approach. This will also uncover any missing `default:` cases.

Reply:
I applied the maximum value approach, but only used it to send warnings, so I hope it is future-proof now.